src/base/iconprovider.cpp

```diff
@@ -57,7 +72,20 @@ IconProvider *IconProvider::instance()
 
 QString IconProvider::getIconPath(const QString &iconId) const
 {
-    return ":/icons/qbt-theme/" + iconId + ".png";
+    const auto it = m_iconFileNames.find(iconId);
+    if (it != m_iconFileNames.end()) {
+        return it.value();
+    }
+
+    const QString iconThemeDir = QLatin1String(":/icons/qbt-theme/");
+    const QString fileSuffix = QLatin1String("256/") + iconId + QLatin1String(".png");
+    for (const auto dir : THEME_DIRS) {
+        const QString probe = iconThemeDir + dir + fileSuffix;
+        if (QFileInfo::exists(probe)) {
+            return m_iconFileNames[iconId] = probe;
```

Comment:
Because I always confuse the order: Are you sure this will `return` the value after the assignment and not the old one?

Reply:
Sure, otherwise `a = b = c` would not work.